### PR TITLE
Add support for new Steam authentication

### PR DESCRIPTION
**Pull request by yaakov-h**

### CODE REVIEW

**Comment by xPaw** [comment]
Can probably drop SentryFileHash/accepting machine auth with this.

File: DepotDownloader/Steam3Session.cs [hunk 1]
```diff
@@ -508,7 +488,37 @@ private void ConnectedCallback(SteamClient.ConnectedCallback connected)
...
             else
             {
-                Console.Write("Logging '{0}' into Steam3...", logonDetails.Username);
+                Console.WriteLine("Logging '{0}' into Steam3...", logonDetails.Username);
+
+                if (logonDetails.Username != null && logonDetails.Password != null && logonDetails.AccessToken is null)
+                {
+                    try
+                    {
+                        var session = await steamClient.Authentication.BeginAuthSessionViaCredentialsAsync(new SteamKit2.Authentication.AuthSessionDetails
```
[the remark this answers is not in the document]
Comment (yaakov-h) [reply DepotDownloader/Steam3Session.cs:497]: Do you have a code example for that you can share?

[the remark this answers is not in the document]
Comment (yaakov-h) [reply DepotDownloader/Steam3Session.cs]: I _think_ I have it working.

**Comment by yaakov-h** [comment]
I haven't had a chance to go over it with a fine-toothed comb but this seems to be working well enough, if it breaks then I'm happy to take bug reports as they come in.

File: DepotDownloader/Steam3Session.cs [hunk 2]
```diff
@@ -553,14 +639,14 @@ private void LogOnCallback(SteamUser.LoggedOnCallback loggedOn)
         {
             var isSteamGuard = loggedOn.Result == EResult.AccountLogonDenied;
             var is2FA = loggedOn.Result == EResult.AccountLoginDeniedNeedTwoFactor;
-            var isLoginKey = ContentDownloader.Config.RememberPassword && logonDetails.LoginKey != null && loggedOn.Result == EResult.InvalidPassword;
+            var isAccessToken = ContentDownloader.Config.RememberPassword && logonDetails.AccessToken != null && loggedOn.Result == EResult.InvalidPassword; // TODO: Get EResult for bad access token
```
Comment (xPaw) [inline DepotDownloader/Steam3Session.cs:642]: Codes I've seen are InvalidSignature and AccessDenied.

**Comment by ABaumher** [comment]
I probably don't belong here, but while i was testing your code, i noticed a minor error at
https://github.com/SteamRE/DepotDownloader/blob/0112ca749d88f9d02f2740495386aa609f6e2729/DepotDownloader/Steam3Session.cs#L679
You provide 5 arguments to string.format, the formatted string only has 4. Can be fixed by adding `{4}`.
```Console.WriteLine("Got Machine Auth: {0} {1} {2} {3} {4}", machineAuth.FileName, machineAuth.Offset, machineAuth.BytesToWrite, machineAuth.Data.Length, hash);```

**Comment by xPaw** [comment]
@ABaumher fixed that in 0e8b6627159cb5ae378d0dac3688b0585d3740d1.

I merged this PR so it can be tested by users, and improvements can be made in further PRs.

**Comment by GoTeamScotch** [comment]
Just to clarify, does this allow the use of -remember-password (referenced by #397)?

Trying out v2.5.0 tonight and I seem to be getting multiple Steam Guard verification code requests.

Example:
```
DepotDownloader.exe -app 223100 -depot 223101 -username GoTeamScotch -remember-password
DepotDownloader.exe -app 223100 -depot 223102 -username GoTeamScotch
```
leads to multiple requests for the account password, then a subsequent Steam Guard code via email.

**Pull request merged**